sched/group/group_create.c

```diff
@@ -119,20 +121,42 @@ int group_initialize(FAR struct task_tcb_s *tcb, uint8_t ttype)
 
   DEBUGASSERT(tcb && !tcb->cmn.group);
 
+  ttype &= TCB_FLAG_TTYPE_MASK;
+
+  /* Late coming kthread reuses the shared group */
+
+  if (ttype == TCB_FLAG_TTYPE_KERNEL && g_kthread_group.tg_info)
+    {
+      tcb->cmn.group = &g_kthread_group;
+      return OK;
+    }
+
+  if (!g_kthread_group.tg_info)
+    {
+      sinfo("sizeof(*group)=%zu\n", sizeof(*group));
+    }
+
   /* Allocate the group structure and assign it to the TCB */
 
-  group = &tcb->group;
+  group = (ttype == TCB_FLAG_TTYPE_KERNEL) ? &g_kthread_group :
+            kmm_zalloc(sizeof(*group));
```

Comment:
```suggestion

```

I have an extreme requirement for the initialization of task structure. All resources must be deterministic in nxtask_init(), which means that there cannot be any access to the memory allocator, Currently in flat mode, all resources except tcb/stack could be configured statically. could you implement this PR based on this premise?

Reply:
Maybe that is better addressed by a separate set of statical task/group management APIs?

Reply:
maybe it's better to map kernel thread to tcb_s not task_tcb_s, and point to the global g_kthread_group

Reply:
What would you choose if there were functional safety and reduced resources? Adding an API will make the system more chaotic, which may not be the desired outcome for everyone. Perhaps we can follow your suggestion to add a group config to implement this feature, but what I'm not sure is, could all kthreads use the same group?